framework/aster-frame/src/vm/options.rs

```diff
@@ -14,6 +14,7 @@ pub struct VmAllocOptions {
     nframes: usize,
     is_contiguous: bool,
     uninit: bool,
+    need_dealloc: bool,
```

Comment:
While implementing frame metadata, I think there's no need to expose the `need_dealloc` property. Just forget about a handle you have a permanent frame that is not recycled:

```rust
let frame = VmAllocOptions::new(1).alloc();
// Write some permanent value to the frame.
core::mem::forget(frame);
```